alacritty/src/config/window.rs

```diff
@@ -108,27 +99,44 @@ impl WindowConfig {
     pub fn maximized(&self) -> bool {
         self.startup_mode == StartupMode::Maximized
     }
-}
 
-impl Default for WindowConfig {
-    fn default() -> WindowConfig {
-        WindowConfig {
-            dimensions: Default::default(),
-            position: Default::default(),
-            padding: Default::default(),
-            decorations: Default::default(),
-            dynamic_padding: Default::default(),
-            startup_mode: Default::default(),
-            class: Default::default(),
-            embed: Default::default(),
-            gtk_theme_variant: Default::default(),
-            title: default_title(),
-            dynamic_title: Default::default(),
+    #[inline]
+    #[cfg(not(any(target_os = "macos", windows)))]
+    pub fn instance(&self) -> &str {
+        match &self.class {
+            Class::Just(instance) | Class::WithGeneral { instance, .. } => instance.as_str(),
+        }
+    }
+
+    #[inline]
+    pub fn set_instance(&mut self, instance: String) {
+        match &mut self.class {
+            Class::Just(i) | Class::WithGeneral { instance: i, .. } => *i = instance,
+        }
+    }
+
+    #[inline]
+    #[cfg(not(any(target_os = "macos", windows)))]
+    pub fn general(&self) -> &str {
+        match &self.class {
+            Class::Just(_) => DEFAULT_NAME,
+            Class::WithGeneral { general, .. } => general.as_str(),
+        }
+    }
+
+    #[inline]
+    pub fn set_general(&mut self, general: String) {
+        match &mut self.class {
+            Class::Just(instance) => {
+                let instance = instance.clone();
+                self.class = Class::WithGeneral { instance, general };
+            },
+            Class::WithGeneral { general: g, .. } => *g = general,
         }
     }
```

Comment:
I'd say this is probably the worst "regression" when it comes to code quality with the new deserializer.

Unfortunately I'm not sure about a good way to abstract over untagged enums with struct and tuple variants. So I'm currently uncertain how I would improve this specific part.